.github/workflows/deploy.yml

```
@@ -146,10 +146,12 @@ jobs:
       - name: Set image output
         id: set-image
         env:
-          ECR_REGISTRY: ${{ vars.ECR_REGISTRY }}
-          ECR_REPOSITORY: ${{ vars.ECR_REPOSITORY }}
+          GHCR_IMAGE: ${{ env.GHCR_IMAGE }}
           GITHUB_SHA: ${{ github.sha }}
-        run: echo "image=${ECR_REGISTRY}/${ECR_REPOSITORY}:${GITHUB_SHA}" >> $GITHUB_OUTPUT
+        # ECS now deploys the GHCR image (same immutable :<sha> the build job pushed
+        # to GHCR alongside ECR). ECR is still pushed for cache/rollback; the deploy
+        # target is GHCR. The composite action injects repositoryCredentials for it.
+        run: echo "image=${GHCR_IMAGE}:${GITHUB_SHA}" >> $GITHUB_OUTPUT
```

actionlint (1.7.12)

[error] shellcheck reported issue in this script: SC2086:info:1:45: Double quote to prevent globbing and word splitting

(shellcheck)